src/ecs/plugins/console.rs

```diff
@@ -0,0 +1,152 @@
+  PlayerSpeed { speed: f32 },
+}
+impl Plugin for ConsolePlugin {
+  fn build(&self, app: &mut App) {
+    app
+      .add_system_to_stage(CoreStage::PreUpdate, (debug_console).after(ImguiFrameSystem))
+      .add_event::<ConsoleCommandEvents>();
+  }
+}
+pub struct ConsoleText {
+  text: String,
+  level: Level,
+}
+pub fn debug_console(
+  imgui: NonSendMut<ImguiState>,
+  mut window: ResMut<Windows>,
+  debug_console_opened: ResMut<ConsoleMenuOpened>,
+  big_font: NonSend<BigFont>,
+  mut items: Local<Vec<ConsoleText>>,
+  mut console_event_writer: EventWriter<ConsoleCommandEvents>,
+) {
+  if !debug_console_opened.0 {
+    return;
+  }
+
+  let active_window = window.get_primary_mut().unwrap();
+  let ui = imgui.get_current_frame();
+  fn add_log(items: &mut Vec<ConsoleText>, log: &str, m_level: Level) {
+    let ct = ConsoleText {
+      text: log.to_string(),
+      level: m_level,
+    };
+    items.push(ct);
+  }
+  imgui::Window::new("Debug Console")
+    .position([0 as f32, 0 as f32], Condition::Once)
+    .size([active_window.width(), 300.0], Condition::Once)
+    .movable(false)
+    .collapsible(false)
+    .resizable(false)
+    .title_bar(false)
+    .focus_on_appearing(true)
+    .build(ui, || {
+      let _f = ui.push_font(big_font.0);
+
+      let cwindow = imgui::ChildWindow::new(0)
+        .border(false)
+        .draw_background(true)
+        .size([active_window.width(), 250.0])
+        .begin(ui);
+      match cwindow {
+        None => {}
+        Some(_) => {
+          for item in items.iter() {
+            const RED: [f32; 4] = [1.0, 0.0, 0.0, 1.0];
+            const GREEN: [f32; 4] = [0.0, 1.0, 0.0, 1.0];
+            const YELLOW: [f32; 4] = [1.0, 1.0, 0.0, 1.0];
+            let item_text = &item.text;
+            let item_level = &item.level;
+
+            let token;
+            match item_level {
+              &Level::ERROR => token = ui.push_style_color(StyleColor::Text, RED),
+              &Level::INFO => token = ui.push_style_color(StyleColor::Text, GREEN),
+              &Level::WARN => token = ui.push_style_color(StyleColor::Text, YELLOW),
+              &Level::TRACE => token = ui.push_style_color(StyleColor::Text, YELLOW),
+              &Level::DEBUG => token = ui.push_style_color(StyleColor::Text, YELLOW),
+            }
+            ui.text(item_text);
+            token.pop();
+          }
+        }
+      }
+      drop(cwindow);
+      let mut buff0 = String::with_capacity(32);
+      let input_width = ui.push_item_width(active_window.width() - 10 as f32);
+      if ui.is_window_focused() && !ui.is_any_item_focused() && !ui.is_mouse_clicked(imgui::MouseButton::Left) {
+        ui.set_keyboard_focus_here();
+      }
+      if ui
+        .input_text("##", &mut buff0)
+        .enter_returns_true(true)
+        .auto_select_all(true)
+        .allow_tab_input(true)
+        .hint("input commands")
+        .build()
+      {
+        fn get_bool_command(command: &str) -> bool {
```

Comment:
what the fuck, just do == "1" inline or whatever

Reply:
This might have a reason for existence but i'd rather it be some generic bool::try_from or whatever idk.